### PR TITLE
Remove some unnecessary mocking from the unit tests

### PR DESCRIPTION
**Pull request by richvdh**
As far as I can tell, all this mocking out of `federation_resource`, `http_client` and `keyring` is totally unnecessary, so we may as well get rid of it.

(Advise looking at the individual commits here: the first commit removes mocking from the test classes; the others remove the support code that was used to set up those mocks and is no longer.)

### CODE REVIEW

File: tests/api/test_filtering.py [hunk 1]
```diff
@@ -42,21 +40,9 @@ def MockEvent(**kwargs):
 
 
 class FilteringTestCase(unittest.TestCase):
-    @defer.inlineCallbacks
```
Comment (richvdh) [inline tests/api/test_filtering.py:45]: `setup_test_homeserver` doesn't return a Deferred.

File: tests/handlers/test_profile.py [hunk 2]
```diff
@@ -44,8 +44,6 @@ def register_query_handler(query_type, handler):
 
         hs = yield setup_test_homeserver(
```
Comment (clokep) [inline tests/handlers/test_profile.py:45]: I don't think we need to `yield` here as you mentioned above.

**Review by clokep — APPROVED** [review]
Seems good if tests pass.

File: tests/handlers/test_profile.py (hunk 2 above)
Comment (richvdh) [reply tests/handlers/test_profile.py:45]: true, but that can be fixed another time :innocent:

**Pull request merged**